docs/SILProgrammersManual.md

```diff
@@ -159,6 +159,165 @@ rules out PartialApplies is no excuse to conflate applied arguments
+
+`AccessPath` contains both a storage location and the "access path"
+within that memory object. Applying an access path to the base of a
+formal access determines the address of the memory operation. The
+access path is a string of index offsets and subobject projection
+indices.
+
+Examples of Def->Use: (Path indices)
+```
+   struct_element_addr %base #1: (#1)
+   ref_tail_addr %ref -> index_addr #1 -> struct_element_addr #2: (@1, #2)
+```
+
+Note that a projection from a reference type to the object's property
+or tail storage is not part of the access path because it is already
+identified by the storage location.
+
+Offset indices are all folded into a single index at the head of the
+path. A missing index corresponds to a zero offset. Offsets that are
+not static constants are still valid but are labeled "unknown". Indexing
+within a subobject is an ill-formed access, but is handled
+conservatively since this rule cannot be enforced.
+
+For example, the following is an invalid access path:
+```
+struct_element_addr %base #1 -> index_addr #1: (INVALID)
+```
+
+The following APIs determine whether an access path contains another
+or may overlap with another.
+
+`AccessPath::contains(AccessPath subPath)`
+
+`AccessPath::mayOverlap(AccessPath otherPath)`
+
+These are extremely light-weight APIs that, in the worst case, require
+a trivial linked list traversal with single pointer comparison for the
+length of subPath or otherPath.
+
+Subobjects are both contained with and overlap with their parent
+storage. An unknown offset does not contain any known offsets but
+overlaps with all offsets.
+
+### Access path def-use relationship
+
+Computing `AccessedStorage` and `AccessPath` for any given SIL address
+involves a use-def traversal to determine the origin of the
+address. It may traverse operations on address, pointer, and reference
+types. The logic that formalizes which SIL operations may be involved
+in the def-use chain is encapsulated with the
+`AccessUseDefChainVisitor`. The traversal can be customized by
+implementing this visitor.
+
+Access def-use chains are divided by three points: the "root", the
+formal access "base", and the memory address.
+
+Def-use example:
+```
+  %root    = alloc_ref $C
+  %base    = ref_element_addr %root : $C, #C.prop
+  %access  = begin_access [read] [static] %base : $*S
+  %memaddr = struct_element_addr %access : $*S, #.field
+  %value   = load [trivial] %memaddr : $*Int64
+  end_access %access : $*S
+```
+
+The first part of the def-use chain computes the formal access base
+from the root of the object (e.g. alloc_ref -> ref_element_addr ->
+begin_access). Reference types are only allowed in this part. Each
+object property and its tail storage is considered a separate formal
+access base. Consequently the reference root is encapsulated within
+the `AccessedStorage` location, and apart from identifying the
+storage, projections from a reference type play no part in the
+AccessPath components.
```

Comment:
"... and apart from identifying the
 storage, projections from a reference type play no part in the
 AccessPath components."
I don't understand this sentence.

Reply:
This sentence still exists in the rewritten documentation "Apart from identifying the storage location, projections from a reference type are not part of the access path." But I don't know how to say it better than that and it should be clear from the context and examples.